src/main/java/uk/gov/pay/products/service/GatewayAccountUpdater.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.pay.products.service;
+
+import com.google.inject.persist.Transactional;
+import uk.gov.pay.products.model.GatewayAccountRequest;
+import uk.gov.pay.products.model.Product;
+import uk.gov.pay.products.persistence.dao.ProductDao;
+import uk.gov.pay.products.persistence.entity.ProductEntity;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GatewayAccountUpdater {
+
+    private final ProductDao productDao;
+
+    @Inject
+    public GatewayAccountUpdater(ProductDao productDao) {
+        this.productDao = productDao;
+    }
+
+    @Transactional
+    public List<Product> doPatch(Integer gatewayAccountId, GatewayAccountRequest gatewayAccountRequest) {
+        List<ProductEntity> productEntities = productDao.findByGatewayAccountId(gatewayAccountId);
```

Comment:
Rather than loading all entities and looping update queries can't we think of a one update query affecting all rows without loading anything? That'd be more efficient

Reply:
Will address it in part 2